### ISSUE TITLE
Bot for Contributor Role

### ISSUE THREAD

**Issue by rlmoser99** [TheOdinProject/top-meta#251]
| Title | Author | Date |
| --- | --- | --- |
| Bot for Contributor Role | `Rachel` | Nov 22 |

# Bot for Contributor Role

## Summary

Have a bot automatically give a role to a person that had __ contributions on any of our repos? (role like "contributor" or "regular contributor")

## Motivation

I had this idea as a way to see those that are contributing to our curriculum & might be good fit to be a maintainer, that we might be overlooking.

## Suggested implementation

A possibility to ask users to submit their GitHub Ids. We already store their discord ids and points on the website's db, right? We can create a new github_id field on there or something.

From there, we could maybe create a custom GitHub action that sends necessary data to @odin-bot whenever a PR is merged or some other event

May need to use GitHub's API to get this data
`https://api.github.com/repos/TheOdinProject/theodinproject/contributors`

Maybe we would need to use 2 fields.
github_username
github_contributions

Or combine them into one `contributions: { github_username: <username> count: <number> }`

## Drawbacks

- Need to add data to our database
- might need to setup OAuth

## Alternatives

We could monthly run a report (using Github's API) and update any newer contributors, or maybe we could give out this role when we notice that someone has produced a few meaningful contributions.

**Comment by rlmoser99** [comment]
With @ChargrilledChook suggestion in Discord to look at https://github.com/TheOdinProject/curriculum/graphs/contributors?from=2021-11-12&to=2022-11-12&type=c

Maybe we should do this manually for the time being. So, I think we need to discuss the number of contributions in a year that would qualify someone for the `contributor` role.

Do we want to talk about a certain number of contributions publicly on this issue, or should we discuss that privately?

**Comment by ChargrilledChook** [comment]
> Maybe we should do this manually for the time being

Agreed. Maybe a standing point on the monthly agenda to check this?
